### PR TITLE
Fix tests for API28-33

### PR DESCRIPTION
**Pull request by LinX64**
Most of the tests were failing because an external session was being followed and never unfollowed.
Also, I've added a few improvements to it; for ex; Running the tests based on the Names order.

### CODE REVIEW

File: app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt [hunk 1]
```diff
@@ -424,13 +445,15 @@ class SearchFollowTest {
 
         onView(withId(R.id.places_autocomplete_search_bar))
             .perform(typeText(place))
+
+        Thread.sleep(3500)
+        // We have to wait for the list to be shown and to be retrieved from the server, otherwise tests will fail!
```
Comment (czorek) [inline app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt:450]: Can't we mock this somehow, do we really need to hit the server for data in tests? In general I think adding sleeps is usually not very reliable anyway

Comment (LinX64) [reply app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt:450]: Good point!
The thing is, we're loading data from Google Places API right after searching for a location. We cannot be sure how long it takes the data to be shown, so this was what I had in my mind.

Another approach is to check if the recyclerView is empty or not. I gave it a try and couldn't check if data is available or not, but I'll try to provide a check for it.

Comment (czorek) [reply app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt:450]: Sure, but isn't it possible to just not actually hit the external API but just mock the data?

Comment (LinX64) [reply app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt:450]: Sure we can, but as far as I understood, Kirill wrote these tests to have the same situation for different devices and to catch unexpected situations, but we can actually use dummy data for it, that's also a good idea.

Comment (czorek) [reply app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt:450]: Can we even check the behavior of different devices by running tests? How does that work?

Comment (LinX64) [reply app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt:450]: We can check most of the things on the Firebase Test lab video section and logs from the testings.

File: app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt [hunk 2]
```diff
@@ -290,26 +286,47 @@ class SearchFollowTest {
...
+        searchActivityScenario.close()
+    }
+
+    private fun unFollowFollowedSessionFromTheFollowingTab() {
+        goToMainScreen()
+
+        expandCard()
+
+        onView(withId(R.id.unfollow_button))
+            .perform(click())
```
Comment (xameYeoh) [inline app/src/androidTest/java/pl/llp/aircasting/SearchFollowTest.kt:329]: Discussed IRL: 
This is not related to unfollowing a session, as it only checks the state of the follow button **on the bottomsheet**, with the OpenAQ user logged in (so all the OpenAQ sessions should appear as your own)

**Comment by LinX64** [comment]
I'll work on this and will check it again, so that we'll be able to merge it ASAP.
